pages/dashboard/profile/[id]/index.tsx

```diff
@@ -14,18 +19,28 @@ interface InitialProps {
 
 const ProfilePage: NextPage<InitialProps> = () => {
   const idStr = useRouter().query.id as string
-  const id = parseInt(idStr, 10)
+  const userId = parseInt(idStr, 10)
 
-  const { data, loading, error } = useProfileQuery({
-    variables: { userId: id },
+  const {
+    data: userData,
+    loading: loadingCurrentUser,
+    error: currentUserError,
+  } = useCurrentUserQuery()
+  const { data: profileData, loading: loadingProfile, error: profileError } = useProfileQuery({
+    variables: { userId },
   })
 
+  const isLoading = loadingCurrentUser || loadingProfile
+  const hasError = currentUserError || profileError
+
   return (
-    <LoadingWrapper loading={loading} error={error}>
+    <LoadingWrapper loading={isLoading} error={hasError}>
       <DashboardLayout>
-        { data?.userById && data?.posts && (
-          <Profile user={data?.userById} posts={data?.posts} />
-        )}
+        <Profile
+          isLoggedInUser={userData?.currentUser?.id === userId}
+          user={profileData?.userById as UserType}
+          posts={profileData?.posts as PostType[]}
```

Comment:
Since the loading wrapper handles not showing any data if `loading` or `error` is falsey, we can use `as Type` to tell TypeScript that the value will be there, which simplifies having to do `data?.x && <content>`